semgrep-core/src/core/Pattern_match.ml

```diff
@@ -36,6 +36,27 @@
  * from a pattern:, we should not merge them!
  *)
 
+type tainted_tokens = Parse_info.t list [@@deriving show]
+
+(* Simplified version of Taint.source_to_sink meant for finding reporting *)
+type taint_call_trace =
+  (* A direct match *)
+  | Toks of Parse_info.t list
+  (* An indirect match through a function call *)
+  | Call of {
+      call_toks : tainted_tokens;
+      intermediate_toks : tainted_tokens;
```

Comment:
what are those intermediate_toks?

Reply:
maybe an example would help?
like with code like  foo(source);   and separately  function foo(x) { sink(source) } 
what would be the taint_call_trace?

Reply:
and what would be those intermediate_toks?

Reply:
@IagoAbal would probably be better at answering this than me. They are modeled after this: https://github.com/returntocorp/semgrep/blob/e3e87a473a8603acad6c7b5fb40edcd47fb6c610/semgrep-core/src/tainting/Taint.ml#L24

According to a separate comment from @IagoAbal on this PR, this probably can't actually happen in Semgrep in practice, and only occurs in DeepSemgrep.

Reply:
Right now it could be called `intermediate_vars` 🤔 those tokens are from intermediate variables that get tainted.